packages/shared/src/lib/interest/RegisterInterest.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+	import type { Interest } from '$lib/interest/intrestStore';
+
+	type Props = {
+		interest: Interest;
+		reference?: HTMLElement;
+		onlyInView?: boolean;
+	};
+
+	const { interest, reference: ref, onlyInView }: Props = $props();
+
+	let inView = $state(false);
+
+	$effect(() => {
+		if (ref && onlyInView) {
+			inView = false;
+
+			const observer = new IntersectionObserver(
+				(entries) => {
+					inView = entries[0].isIntersecting;
+				},
+				{
+					root: null
+				}
+			);
+
+			observer.observe(ref);
+
+			return () => {
+				inView = false;
+				observer.disconnect();
+			};
+		}
+	});
+
+	$effect(() => {
+		const shouldSubscribe = !onlyInView || inView;
+		if (interest && shouldSubscribe) {
```

Comment:
Variable `interest` will according to types never be falsy.

Reply:
Correct. At one point in time it was possibly undefined, but that is no longer true.